File: programmers/level2/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    left = 0
    right = len(people) - 1

    while left <= right:
        temp = people[left] + people[right]

        if temp <= limit:
            left += 1
            right -= 1

        elif temp > limit:
            right -= 1

        answer += 1

    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int sum = 0;
        Arrays.sort(people);
        int left = 0, right = people.length - 1;

        while(left <= right) {
            if(people[left] + people[right] <= limit)
                left++;
            right--;
            answer++;
        }
        
        return answer;
    }
}